### ISSUE TITLE
Stream end event don't been called

### ISSUE THREAD

**Issue by herlon214** [tomas/needle#139]
Hi guys, here is the code:

```javascript
var stream = needle.get(url, {headers: headers, "follow_max": 5, compressed: true}).pipe(out);

    stream.on('end', function() {
      console.log("Done");
    });
```

The console never show "Done", why?
Thanks in advance.

**Comment by tomas** [comment]
What content type are you downloading? Are you sure there aren't any errors occurring?

**Comment by herlon214** [comment]
I fixed putting the pipe on the other line:

```javascript
var stream = needle.get(url, {headers: headers, "follow_max": 5});
    stream.pipe(out);
    stream.on('end', function() {
      extract();
    });
```

**Comment by tomas** [comment]
The fix that closes #140 should also take care of this. Can you check again?
